addons/sale_contract/models/sale_contract.py

```diff
@@ -483,6 +511,27 @@ def fill_remind(key, domain, write_pending=False):
 
         return True
 
+    def _cron_set_pending_state(self, cr, uid, context=None):
+        today = fields.date.context_today(self, cr, uid, context=context)
+
+        # find contracts with overdue invoices
+        overdue_account_ids = []
+        acc_analytic_ids = self.search(cr, uid, [('state', '=', 'open')], context=context)
+        for account in self.browse(cr, uid, acc_analytic_ids, context=context):
+            overdue_domain = [('invoice_line_ids.account_analytic_id', '=', account.id), ('state', '=', 'open'), ('date_due', '<', today)]
+            invoice_num = self.pool['account.invoice'].search_count(cr, uid, overdue_domain)
```

Comment:
please don't, this task was ok and just needed a few cleaning because of the renaming of account_analytic_analysis, no full refactoring